pypose/module/dynamics.py

```diff
@@ -0,0 +1,158 @@
+
+    def observation(self):
+        pass
+
+    def reset(self,t=0):
+        self.t.fill_(0)
+
+    @property
+    def A(self):
+        if hasattr(self, '_A'):
+            return self._A
+        else:
+            func = lambda x: self.state_trasition(x, self.input)
+            return jacobian(func, self.state, **self.jacargs)
+
+    @property
+    def B(self):
+        if hasattr(self, '_B'):
+            return self._B
+        else:
+            func = lambda x: self.state_trasition(self.state, x)
+            return jacobian(func, self.input, **self.jacargs)
+
+    @property
+    def C(self):
+        if hasattr(self, '_C'):
+            return self._C
+        else:
+            func = lambda x: self.observation(x, self.input)
+            return jacobian(func, self.state, **self.jacargs)
+ 
+    @property
+    def D(self):
+        if hasattr(self, '_D'):
+            return self._D
+        else:
+            func = lambda x: self.observation(self.state, x)
+            return jacobian(func, self.input, **self.jacargs)
+
+
+class LTI(_System):
+    r'''
+    A sub-class of '_System' to represent the dynamics of discrete-time Linear Time-Invariant (LTI) system.
+    
+    Args:
+        A, B, C, D (:obj:`Tensor`): The coefficient matrix in the state-space equation of LTI system,
+        c1 (:obj:`Tensor`): The constant input of the system,
+        c2 (:obj:`Tensor`): The constant output of the system.
+        
+    Note:
+        According to the actual physical meaning, the dimensions of A, B, C, D must be the consistent,
+        whether in batch or not.
+    '''
+    
+    def __init__(self, A, B, C, D, c1=None, c2=None):
+        super(LTI, self).__init__(time=False)
+        assert A.ndim == B.ndim == C.ndim == D.ndim, "Invalid System Matrices dimensions"
+        self._A, self._B, self._C, self._D = A, B, C, D
+        self._c1, self._c2 = c1, c2
+
+    @property
+    def c1(self):
+        return self._c1
+    
+    @property
+    def c2(self):
+        return self._c2
+    
+    def forward(self, state, input):
+        r'''
+        Args:
+            state (:obj:`Tensor`): The state of the current timestamp of LTI system,
+            input (:obj:`Tensor`): The input of the current timestamp of LTI system.
+
+        Return:
+            Tuple of Tensors: The state of the next timestamp (state-transition) and the system output (observation).
+
+            
+        Every linear time-invariant lumped system can be described by a set of equations of the form
+        which is called the state-space equation.
+        
+        .. math::
+            \begin{align*}
+                \mathbf{z} = \mathbf{A}\mathbf{x} + \mathbf{B}\mathbf{u} + \mathbf{c}_1 \\
+                \mathbf{y} = \mathbf{C}\mathbf{x} + \mathbf{D}\mathbf{u} + \mathbf{c}_2 \\
+            \end{align*}
+
+        where we use :math:`\mathbf{x}` and :math:`\mathbf{u}` to represent state and input of the current timestamp of LTI system.
+            
```

Comment:
1. Will it be better if moving this equation and explanation to the constructor of this class? Then the forward function only needs to provide examples. 
2. The arguments of the constructor will be self-explained also.